### ISSUE TITLE
Prevent pairs with mentor = learner

### ISSUE THREAD

**Issue by djmitche** [mozilla/mentoring#47]
> I was looking for a ``mentor != learner`` test, but it looks like the UI enforces that at the moment. Model-level enforcement of ``learner != mentor`` is tricky. The method [clean](https://docs.djangoproject.com/en/3.1/ref/models/instances/#django.db.models.Model.clean) can have the logic, and will be called when you save objects in forms like the Django admin. Django Rest Framework does not call  ``clean()`` or ``.full_clean()``, and wants [object-level validation](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation) in the ``.validate()`` method of the serializer. ``clean()`` and ``full_clean()`` are not called when calling ``save`` directly in code, or when doing bulk updates. If you want model-level safety everywhere, you end up with a lot of duplicate code.

_Originally posted by @jwhitlock in https://github.com/mozilla/mentoring/pull/46#discussion_r564594018_

**Comment by djmitche** [comment]
It looks like I could use a [CheckConstraint](https://docs.djangoproject.com/en/3.1/ref/models/constraints/) for this?

**Comment by jwhitlock** [comment]
Yes, a ``CheckConstraint`` should work! You'll get a nasty failure (an ``IntegrityError``) if it ever fails, and then you can decide if it is worth turning it into a validation error.
